### PR TITLE
Concretify cops model

### PR DESCRIPTION
**Pull request by jcorbin**
And drop the unused interface chaff that's been blowing up my linter for too long (due to lol-no-docs).

### CODE REVIEW

File: internal/cops/display/model.go [hunk 1]
```diff
@@ -5,19 +5,14 @@ import (
 )
 
 // Model is the interface for a terminal color rendering model.
-type Model interface {
-	// RenderRGBA appends the ANSI sequence for changing the foreground and
-	// background color to the nearest colors supported by the terminal color
-	// model.
-	RenderRGBA(buf []byte, cur Cursor, fg, bg color.RGBA) ([]byte, Cursor)
```
Comment (kriskowal) [inline internal/cops/display/model.go]: Let’s go all the way—

```go
type Model func(buf []byte, cur Cursor, fg, bg color.RGBA) ([]byte, Cursor)

func model(fg, bg func([]byte, color.RGBA) []byte) Model {
    return func(fg, bg func([]byte, color.RGBA) []byte {
        // ...
    }
}

var Model0 = model(renderNoColor, renderNoColor)
// ...
```

So third-party models are still possible, without needing an interface, not precluded by a private struct.

Comment (jcorbin) [reply internal/cops/display/model.go]: I really don't see the utility for "third party modules" in this internal package... that said I'm cool with your `Model` func type, just not the convenience combinator... let's try that, as I suspect it'll have a small perf win (where-as the func-func would likely be either a stay-in-place or small loss).

Comment (kriskowal) [reply internal/cops/display/model.go]: :+1: The convenience wrapper does come with a cost. There is an opportunity to merge the foreground and background colors into a single CSI sequence.

**Comment by jcorbin** [comment]
Future work can unify the CSI; this fixed the spurious reset when using `Model0` that was lurking.

File: internal/cops/display/ansi.go [hunk 2]
```diff
@@ -31,56 +31,38 @@ func init() {
...
-	if i, ok := colorIndex[c]; ok {
-		return append(buf, fgColorStrings[i]...)
+func renderNoColor(buf []byte, cur Cursor, _, _ color.RGBA) ([]byte, Cursor) { return buf, cur }
+
+func renderCompatColor24(buf []byte, cur Cursor, fg, bg color.RGBA) ([]byte, Cursor) {
+	if fg != cur.Foreground {
+		if i, ok := colorIndex[fg]; ok {
+			buf = append(buf, fgColorStrings[i]...)
+		} else {
+			buf = renderColor24(append(buf, "\033[38;2"...), fg)
```
Comment (kriskowal) [inline internal/cops/display/ansi.go]: Optional: I’m sure you’ve noticed my propensity for keeping these append lines flat.

Comment (jcorbin) [reply internal/cops/display/ansi.go]: Sure, the nesting pre-dated this PR, I slipped it by you in an earlier one ;-)

**Pull request merged**